src/core/net/netif.hpp

```diff
@@ -239,6 +239,30 @@ class Netif : public InstanceLocator
      */
     otError RemoveUnicastAddress(const NetifUnicastAddress &aAddress);
 
+    /**
+     * This method adds an internal (to OpenThread) unicast address to the network interface.
+     *
+     * @param[in]  aAddress  A reference to the unicast address.
+     *
+     * @retval OT_ERROR_NONE          Successfully added (or updated) the unicast address.
+     * @retval OT_ERROR_INVALID_ARGS  The address indicated by @p aAddress is link-local address or an external address.
+     * @retval OT_ERROR_NO_BUFS       The maximum number of allowed internal addresses are already added.
+     *
+     */
+    otError AddInternalUnicastAddress(const NetifUnicastAddress &aAddress);
```

Comment:
How about instead of adding a new API, we add a paramter to indicate the origin of the address, simple case can be boolean `mIsInternal`. 
Or we can define an enumeration which gives more detail about origin of the address:
```
enum AddressOrigin
{
     kOriginExternal,
     kOriginSlaac,
     kOriginDhcp,
    ...
};
```

Here is my suggestion: 
- We rename existing `AddUnicastAddress()` to `AddPreAllocatedUnicastAddress()`  to indicate address parameter is already allocated (no need to allocate a new entry from the pool). 
- We combine `AddInternalUnicastAddress()` and `AddExternalUnicastAddress()` into:
    - `AddUnicastAddress(const Address &aAddress, AddressOrigin aOrigin)` 
   - (or `AddUnicastAddress(const Address &aAddress,bool aIsInternal)`)


Thoughts?

Reply:
How about even tracking the origin of the address (or at min if it's internal/external) in the `otNetifAddress` struct?

Reply:
Thanks for your suggestions. I will try add parameter to existing methods. However, this PR aims at making syncing addresses with LwIP easier, and I tried not to increase memory usage in `otNetifAddress`. Besides, I think with this implementation, we can tell if a address is internal or external added according to the memory location, but I tend to add those APIs when we use them.